protocol/src/lib.rs

```diff
@@ -783,15 +783,41 @@ impl CipherSession {
     }
 }
 
+/// Fill a slice with random bytes. This trait _should_ be cryptographically secure; however, a
+/// psuedo-random number generator may be sufficient depending on your security model.
+pub trait FillBytes {
+    /// Fill a 32 byte slice with random data.
+    fn fill_bytes(&mut self, dest: &mut [u8; 32]);
+}
+
+#[cfg(feature = "std")]
+macro_rules! impl_fill_bytes {
```

Comment:
When I see macros:

<img width="442" height="296" alt="image" src="https://github.com/user-attachments/assets/81f6ed96-eb7f-487c-a553-ff79eb14b69c" />

But this one seems good since small and contained.